### PR TITLE
(feat) add enhanced adherence in clinical view

### PR DESCRIPTION
**Pull request by gabriel090**
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

adherence

## Screenshots

*None.*

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by makombe** [comment]
@gabriel090  Please resolve the conflicts. Thanks

File: packages/esm-patient-clinical-view-app/src/dashboard.meta.tsx [hunk 1]
```diff
@@ -21,3 +21,12 @@ export const defaulterTracingDashboardMeta = {
   moduleName: '@kenyaemr/esm-patient-clinical-view-app',
   config: {},
 };
+
+export const enhancedAdheranceDashboardMeta = {
+  slot: 'patient-chart-adherence-dashboard-slot',
+  columns: 1,
+  title: 'Adherence',
```
Comment (donaldkibet) [inline packages/esm-patient-clinical-view-app/src/dashboard.meta.tsx:28]: We should have this rename to `enhanced adherence` for UI naming

File: packages/esm-patient-clinical-view-app/src/enhanced-adherence/adherence-table.component.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                  return (
+                    <TableRow key={obsIndex}>
+                      <td>{o?.obsDatetime.split('T')[0]}</td>
+                      <td>{o?.value}</td>
+                      <td>
+                        <OverflowMenu ariaLabel="Actions" size="sm" flipped>
+                          <OverflowMenuItem
+                            hasDivider
+                            //itemText={t('edit', 'Edit')}
+                            // onClick={}
```
Comment (donaldkibet) [inline packages/esm-patient-clinical-view-app/src/enhanced-adherence/adherence-table.component.tsx:31]:
```suggestion

```

File: packages/esm-patient-clinical-view-app/src/enhanced-adherence/adherence-table.component.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  return (
+    <>
+      {encounterData.map(({ encounter }, index) => {
+        const { obs = [] } = encounter ?? {};
+
+        return (
+          <table key={index}>
+            <TableBody>
+              {obs?.map((o, obsIndex) => {
+                if (!uniqueDates.has(o?.obsDatetime) && o?.concept?.uuid === '1639AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA') {
```
Comment (donaldkibet) [inline packages/esm-patient-clinical-view-app/src/enhanced-adherence/adherence-table.component.tsx:20]: should avoid hard coding concepts like this in code

File: packages/esm-patient-clinical-view-app/src/enhanced-adherence/enhanced-adherence.components.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+} from '@carbon/react';
+import { ErrorState } from '@openmrs/esm-patient-common-lib';
+import AdherenceTable from './adherence-table.component';
+type EnhancedAdherenceProps = {
+  patientUuid: string;
+};
+
+const EnhancedAdherence: React.FC<EnhancedAdherenceProps> = ({ patientUuid }) => {
+  const { t } = useTranslation();
+  const headerTitle = t('enhancedadherence', 'Advanced Enhanced Adherence');
```
Comment (donaldkibet) [inline packages/esm-patient-clinical-view-app/src/enhanced-adherence/enhanced-adherence.components.tsx:26]:
```suggestion
  const headerTitle = t('enhancedadherence', 'Enhanced Adherence');
```
to have consistency same as☝🏽

**Review by donaldkibet — CHANGES_REQUESTED** [review]
Thanks @gabriel090 Great work 🚀 let some suggestions.

**Comment by ojwanganto** [comment]
@gabriel090 I think we may need to add more columns to the table provided. Do we capture adherence score anywhere? or anything like final outcome? Kindly work with BA

**Comment by donaldkibet** [comment]
@Gabriel090, I am closing this PR for now as it has been inactive for some time. I believe it would be beneficial to start afresh on the implementation of this feature.

Thank you for your understanding.

**Pull request closed without merging**